media-sound/mumble/mumble-9999.ebuild

```diff
@@ -64,97 +65,41 @@ BDEPEND="
 	virtual/pkgconfig
 "
 
-# NB: qmake does not support multilib but it's fine to configure
-# for the native ABI here
 src_configure() {
-	myuse() {
-		[[ -n "${1}" ]] || die "myuse: No use option given"
-		use ${1} || echo no-${1}
-	}
 
-	local conf_add=(
-		bundled-celt
-		no-bundled-opus
-		no-bundled-speex
-		no-embed-qt-translations
-		no-server
-		no-update
-		$(myuse alsa)
-		$(myuse dbus)
-		$(usex debug 'symbols debug' release)
-		$(myuse g15)
-		$(usex jack '' no-jackaudio)
-		$(myuse opus)
-		$(myuse oss)
-		$(myuse portaudio)
-		$(myuse pulseaudio)
-		$(myuse rnnoise)
-		$(usex speech '' no-speechd)
-		$(usex zeroconf '' no-bonjour)
+	local mycmakeargs=(
+		"-Dbundled-celt=ON"
+		"-Dbundled-opus=OFF"
+		"-Dbundled-speex=OFF"
+		"-Doverlay=ON"
+		"-Dserver=OFF"
+		"-Dupdate=OFF"
+		"-Dalsa=$(usex alsa)"
+		"-DBUILD_TESTING=$(usex test)"
+		"-Ddbus=$(usex dbus)"
+		"-Dg15=$(usex g15)"
+		"-Djackaudio=$(usex jack)"
+		"-Dportaudio=$(usex portaudio)"
+		"-Dpulseaudio=$(usex pulseaudio)"
+		"-Drnnoise=$(usex rnnoise)"
+		"-Dspeechd=$(usex speech)"
+		"-Dtranslations=$(usex nls)"
+		"-Dzeroconf=$(usex zeroconf)"
 	)
 
-	use pch || conf_add+=( no-pch )
-
-	eqmake5 "${S}/main.pro" -recursive \
-		CONFIG+="${conf_add[*]}" \
-		DEFINES+="PLUGIN_PATH=/usr/$(get_libdir)/mumble"
-}
-
-multilib_src_compile() {
-	local emake_args=(
-		# place libmumble* in a subdirectory
-		DESTDIR_ADD="/${MULTILIB_ABI_FLAG}"
-		{C,L}FLAGS_ADD="$(get_abi_CFLAGS)"
-	)
-	# build only overlay library for other ABIs
-	multilib_is_native_abi || emake_args+=( -C overlay_gl )
-	emake "${emake_args[@]}"
-	emake clean
-}
-
-src_compile() {
-	multilib_foreach_abi multilib_src_compile
-}
-
-multilib_src_install() {
-	local dir=$(usex debug debug release)
-	insinto /usr/$(get_libdir)/${PN}
-	doins "${dir}/${MULTILIB_ABI_FLAG}"/libmumble.so*
-	if multilib_is_native_abi; then
-		dobin "${dir}"/mumble
-		doins "${dir}"/libcelt0.so* "${dir}"/plugins/lib*.so*
-	fi
-}
-
-src_install() {
-	multilib_foreach_abi multilib_src_install
-
-	dodoc CHANGES README.md
-	dobin scripts/mumble-overlay
-
-	insinto /usr/share/services
-	doins scripts/mumble.protocol
-
-	domenu scripts/mumble.desktop
-
-	doicon -s scalable icons/mumble.svg
-
-	doman man/mumble-overlay.1
-	doman man/mumble.1
-}
-
-pkg_preinst() {
-	xdg_pkg_preinst
+	cmake_src_configure
 }
 
 pkg_postinst() {
-	xdg_pkg_postinst
+	xdg_icon_cache_update
+	xdg_desktop_database_update
```

Comment:
Please change that back when you use `xdg.eclass` again.

Reply:
Ah, now I remember why.
The ebuild fails if I simply use `xdg` with:
```
FATAL: cmake_src_prepare has not been run
```

That is why I changed it.
So it seems like the xdg eclass messes up.

It also seems like there are quite a few other ebuilds that simply uses the utils class to not have to deal with this either.

Reply:
Just add
```
src_prepare() {
        # required because of xdg.eclass also providing src_prepare
        cmake_src_prepare
}
```
to the ebuild.